core-libs/setup/ssr/engine-decorator/cx-express-engine-decorator.ts

```diff
@@ -15,43 +15,45 @@ import {
 } from '../optimized-engine/ssr-optimization-options';
 import { getServerRequestProviders } from '../providers/ssr-providers';
 
-export type NgExpressEngineInstance = (
+export type CxExpressEngineInstance = (
   filePath: string,
   options: object,
   callback: SsrCallbackFn
 ) => void;
 
-export type NgExpressEngine = (
-  setupOptions: Readonly<NgSetupOptions>
-) => NgExpressEngineInstance;
+export type CxExpressEngine = (
+  setupOptions: Readonly<CommonEngineRenderOptions & CommonEngineOptions>
```

Comment:
[QUESTION]
Why do we mix `CommonEngineRenderOptions` with `CommonEngineOptions`, being our new setup options of our new NgExpressEngine?

`CommonEngineRenderOptions` and  `CommonEngineOptions` are semantically different from each other.
- `CommonEngineOptions` are used to _create and initialize_ `CommonEngine` (with `new CommonEngine(CommonEngineOptions)`
- `CommonEngineRenderOptions` are used only when rendering with the method `render()` of `CommonEngine`

Reply:
In `ngExpressEngine` we need params from both interfaces during the instantiation, but indeed, mixing all of the might be undesirable. We can do it similarly how `@nguniversa` did it:
```ts
export type CxSetupOptions = 
  Pick<CommonEngineRenderOptions, 'providers' | 'publicPath' | 'inlineCriticalCss'> 
& CommonEngineOptions;
```